base/compiler/typelattice.jl

```diff
@@ -220,6 +220,7 @@ widenconst(c::PartialTypeVar) = TypeVar
 widenconst(t::PartialStruct) = t.typ
 widenconst(t::Type) = t
 widenconst(t::TypeVar) = t
+widenconst(t::Core.VarargMarker) = t
```

Comment:
Shouldn't widenconst not be called on one of these?

Reply:
Ideally yes, but currently inference is pretty liberal about calling both this and rewrap_unionall on non-Types and I just wanted to preserve the status quo for now. I'd argue a similar argument could be made for `TypeVar` just above.